### ISSUE TITLE
Add Power to Reactions

### ISSUE THREAD

**Issue by antonkomarev** [cybercog/laravel-love#26]
Possible names:
- Reaction Power
- Reaction Rate
- Reaction Force
- Reaction Strength

### Case 1

Allow Reacter to react to Reactant with same reaction type more than once.

> **Example:**
> In social news application user Jane can vote up article up to 5 times.

Similar to Medium's Claps:
- https://uxdesign.cc/fifty-shades-of-claps-a-case-study-for-clapping-on-medium-22c6e7b71140

### Case 2

Reacter has some kind of authority inside of the  application and his reactions should have more weight among others.

> **Example:**
> In Question & Answer application user Jane solved more than 1000 questions. Jane's reactions will have more power now.

Additionally by this way it will be easier segregate reactions from more notable application users.

**Comment by BenQoder** [comment]
Any Update On This?

**Comment by antonkomarev** [comment]
No. Haven't designed it yet. Not sure it will be done in near future because I'm hardly working on another free open source secret project.
Feel free to discuss this feature and propose implementations in PR. Any kind of help speeds up development process.

**Comment by antonkomarev** [comment]
With reaction power `Reaction` model `getWeight` method should possibly return not just a weight of reaction type, but weight of reaction type multiplied on power.

```php
$reactionType = ReactionType::create([
    'name' => 'Love',
    'weight' => 2,
]);
$power = 3;

$reacter->reactTo($reactant, $reactionType, 3);

$reaction = Reaction::query()
    ->where('reactant_id', $reactant->getId())
    ->where('reacter_id', $reacter->getId())
    ->where('reaction_type_id', $reactionType->getId())
    ->first();

echo $reaction->getWeight(); // 6
```

**Comment by antonkomarev** [comment]
What is a default value of reaction power in all applications which are not using this feature?

`1` or `null`?

**Comment by antonkomarev** [comment]
My have conflicts with Mutually Exclusive Reactions #75

**Comment by BenQoder** [comment]
> With reaction power `Reaction` model `getWeight` method should possibly return weight of reaction type multiplied on power instead of just proxying a weight of reaction type.
> 
> ```
> $reactionType = ReactionType::create([
>     'name' => 'Love',
>     'weight' => 2,
> ]);
> $power = 3;
> 
> $reacter->reactTo($reactant, $reactionType, 3);
> 
> $reaction = Reaction::query()
>     ->where('reactant_id', $reactant->getId())
>     ->where('reacter_id', $reacter->getId())
>     ->where('reaction_type_id', $reactionType->getId())
>     ->first();
> 
> echo $reaction->getWeight(); // 6
> ```
> 
> I'm not sure that 5 likes with power of `2` are equal to 2 likes with power of `5`. With this concept - both of them will produce weight equal to 10, but in fact one person will be able to overrate many people with just one powered like.

Thanks A Lot For Considering This.
You Are Also On The Right Track.
5 likes with the power of `2` is equal to 2 likes with the power of `5` which both case equals 10.
And The Default Power Should Be `1` In Other Applications

**Comment by antonkomarev** [comment]
Default power value `1` gives ability to quick start using Power whenever you want without any future database changes.

**Comment by antonkomarev** [comment]
Another one concept where to use Reaction Power:
> Proactive active users will have more power on their reactions.

For example in Question & Answer application user Jane solved more than 1000 questions. Jane reactions could have more power now.

By this way it will be easier segregate reactions from more notable product users.

**Comment by antonkomarev** [comment]
One more thing to consider. When reacter created a reaction with `power=2` and want to increase\decrease power, how should we do it? `unreactTo` Reactant and `reactTo` it again with new power value?

**Comment by antonkomarev** [comment]
This feature was redesigned to new Weighted Reaction System in #85

**Comment by antonkomarev** [comment]
I'm closing this issue. Core logic was implemented in #91.

It will be released in Laravel Love v8.
